core/headerchain.go

```diff
@@ -461,6 +461,20 @@ func (hc *HeaderChain) GetTd(hash common.Hash, number uint64) *big.Int {
 	return td
 }
 
+// GetS retrieves a block's total entropy in the canonical chain from the
+// database by hash and number, caching it if found.
+func (hc *HeaderChain) GetS(hash common.Hash, number uint64) *big.Float {
+	s := rawdb.ReadS(hc.headerDb, hash, number)
+	return s
+}
+
+// GetDeltaS retrieves a block's change in entropy since the last dom in the canonical chain from the
+// database by hash and number, caching it if found.
+func (hc *HeaderChain) GetDeltaS(hash common.Hash, number uint64) []*big.Float {
+	deltaS := rawdb.ReadDeltaS(hc.headerDb, hash, number)
```

Comment:
Please create an issue for an important optimization:
`s` & `deltaS` should be cached in the header to avoid excessive database access.